docs/user-guides/retrieval.md

```diff
@@ -0,0 +1,191 @@
+  	  with:
+  	  	dim: 512  # input dimension of PCA
+  	  	data_path: workspace
+  	  	n_components: 128  # output dimension of PCA
+  	  metas:
+  	  	py_modules:
+  	  	  - annlite.executor
+```
+
+| Parameter               | Description                                                                                                                  |
+|-------------------------|------------------------------------------------------------------------------------------------------------------------------|
+| `dim`                  | Dimension of embeddings. The output of the encoder as well as the input of PCA.|
+| `n_components` | Output dimension of PCA.|
+
+
+Then you need to train a PCA model:
+
+
+Here is the comparison of memory usage before and after PCA:
+
+```{figure} images/memory_usage_dim_512.png
+:width: 80%
+
+```
+```{figure} images/memory_usage_dim_128.png
+:width: 80%
+
+```
+
+Now the memory usage for indexing 10 millions data decreases from 60GB+ to 30GB, saving more than 50% memory.
+
+However, PCA will definitely lead to information losses since we remove some dimensions. And more dimensions you remove, more information losses will be. So there will be a trade-off between efficiency and accuracy. 
+
+### How do I know whether PCA is needed in my case?
+From our experiments, the memory usage is **linear** to the data size: **1 millions data with dimension of 512 will approximately need 8G-10G**. So you can have an approximately value for memory usage based on your data size.
+
+
+## How to deal with very large dataset?
+For a very large dataset, for example 100M or even 1B, it's not possible to implement index operation on a single machine. **Sharding**, a type of partitioning that separates large dataset into smaller, faster, more easily managed parts, is needed in this case.
+
+You need to speicify the `shards` and `polling` in YAML config:
+```yaml
+jtype: Flow
+version: '1'
+with:
+  port: 51000
+executors:
+  - name: encoder
+    uses:
+      jtype: CLIPEncoder
+      metas:
+        py_modules:
+          - clip_server.executors.clip_torch
+  - name: indexer
+  	uses:
+  	  jtype: AnnLiteIndexer
+  	  with:
+  	  	dim: 512  # input dimension of PCA
+  	  	data_path: workspace
+  	  	n_components: 128  # output dimension of PCA
+  	  metas:
+  	  	py_modules:
+  	  	  - annlite.executor
+  	shards: 5
+  	polling: {'/index': 'ANY', '/search': 'ALL', '/update': 'ALL',
+              '/delete': 'ALL', '/status': 'ALL'}
+```
+
+| Parameter               | Description                                                                                                                  |
+|-------------------------|------------------------------------------------------------------------------------------------------------------------------|
+| `shards`                  | Number of shardings. |
+| `polling` | Polling strategy for different endpoints.|
+
+
+### Why different polling strategies are needed for different endpoints?
```

Comment:
please provide the reference link as well

Reply:
what reference did you mean?

Reply:
ok i see, link for polling strategies